commit d6d6fd7c35e3c14ecbddc69317a65833c41c506b
Author: Jon Jensen <jon@endpoint.com>
Date:   Fri Feb 6 10:40:21 2009 -0700

    Add option hooks.emaildiff to include full diff in post-receive-email.
    
    I've always found that a very important part of receiving email
    notification of commits is being able to read the diff inline,
    easily, where I can reply and quote the diff and make comments.
    It's similar to the reason patches sent to the Git mailing list
    need to be inline, not attachments.
    
    Since post-receive-email didn't have that option, I added it as a
    boolean config variable, hooks.emaildiff.
    
    Signed-off-by: Jon Jensen <jon@endpoint.com>

diff --git a/contrib/hooks/post-receive-email b/contrib/hooks/post-receive-email
index 28a3c0e..47ddfdb 100644
--- a/contrib/hooks/post-receive-email
+++ b/contrib/hooks/post-receive-email
@@ -44,6 +44,9 @@
 #   --pretty %s", displaying the commit id, author, date and log
 #   message.  To list full patches separated by a blank line, you
 #   could set this to "git show -C %s; echo".
+# hooks.emaildiff
+#   If set, then a full diff of changes is sent in addition to the default
+#   summary output.
 #
 # Notes
 # -----
@@ -409,8 +412,13 @@ generate_update_branch_email()
 	# - including the undoing of previous revisions in the case of
 	# non-fast forward updates.
 	echo ""
-	echo "Summary of changes:"
-	git diff-tree --stat --summary --find-copies-harder $oldrev..$newrev
+	if [ -n "$emaildiff" ]; then
+		echo "Summary of changes and diff:"
+		git diff-tree --stat --summary --find-copies-harder -p $oldrev..$newrev
+	else
+		echo "Summary of changes:"
+		git diff-tree --stat --summary --find-copies-harder $oldrev..$newrev
+	fi
 }
 
 #
@@ -666,6 +674,7 @@ announcerecipients=$(git config hooks.announcelist)
 envelopesender=$(git config hooks.envelopesender)
 emailprefix=$(git config hooks.emailprefix || echo '[SCM] ')
 custom_showrev=$(git config hooks.showrev)
+emaildiff=$(git config hooks.emaildiff)
 
 # --- Main loop
 # Allow dual mode: run from the command line just like the update hook, or
